core/include/usBundle.h

```diff
@@ -47,143 +48,297 @@ typedef ServiceReference<void> ServiceReferenceU;
+     * {@code STATE_RESOLVED} state before it can be started. The Framework may
+     * attempt to resolve a bundle at any time.
+     *
+     * The value of {@code STATE_RESOLVED} is 0x00000004.
+     */
+    STATE_RESOLVED = 0x00000004,
+
+    /**
+     * The bundle is in the process of starting.
+     *
+     * A bundle is in the {@code STATE_STARTING} state when its {@link #Start(int)
+     * Start} method is active. A bundle must be in this state when the bundle's
+     * {@link BundleActivator#Start(const BundleContext&)} is called.
+     * If the {@code BundleActivator#Start} method completes without exception, then
+     * the bundle has successfully started and moves to the {@code STATE_ACTIVE}
+     * state.
+     *
+     * If the bundle has a {@link Constants#ACTIVATION_LAZY lazy activation
+     * policy}, then the bundle may remain in this state for some time until the
+     * activation is triggered.
+     *
+     * The value of {@code STATE_STARTING} is 0x00000008.
+     */
+    STATE_STARTING = 0x00000008,
+
+    /**
+     * The bundle is in the process of stopping.
+     *
+     * A bundle is in the {@code STATE_STOPPING} state when its {@link #Stop(int)
+     * Stop} method is active. A bundle is in this state when the bundle's
+     * {@link BundleActivator#Stop(const BundleContext&)} method
+     * is called. When the {@code BundleActivator#Stop} method completes the bundle
+     * is stopped and moves to the {@code STATE_RESOLVED} state.
+     *
+     * The value of {@code STATE_STOPPING} is 0x00000010.
+     */
+    STATE_STOPPING = 0x00000010,
+
+    /**
+     * The bundle is now running.
+     *
+     * A bundle is in the {@code STATE_ACTIVE} state when it has been successfully
+     * started and activated.
+     *
+     * The value of {@code STATE_ACTIVE} is 0x00000020.
+     */
+    STATE_ACTIVE = 0x00000020
+  };
+
+  enum StartOptions : uint32_t {
+
+    /**
+     * The bundle start operation is transient and the persistent autostart
+     * setting of the bundle is not modified.
+     *
+     * This bit may be set when calling {@link #Start(uint32_t)} to notify the
+     * framework that the autostart setting of the bundle must not be modified.
+     * If this bit is not set, then the autostart setting of the bundle is
+     * modified.
+     *
+     * @see #Start(uint32_t)
+     */
+    START_TRANSIENT = 0x00000001,
+
+    /**
+     * The bundle start operation must activate the bundle according to the
+     * bundle's declared
+     * {@link Constants#PLUGIN_ACTIVATIONPOLICY activation policy}.
+     *
+     * This bit may be set when calling {@link #Start(uint32_t)} to notify the
+     * framework that the bundle must be activated using the bundle's declared
+     * activation policy.
+     *
+     * @see Constants#PLUGIN_ACTIVATIONPOLICY
+     * @see #Start(uint32_t)
+     */
+    START_ACTIVATION_POLICY = 0x00000002
+  };
+
+  enum StopOptions {
```

Comment:
I think it would be a good idea to mark these options as not supported. Clients will most likely think there is full support for these options if they are documented.